src/views/TopTenResults.vue

```diff
@@ -43,34 +43,118 @@ import TopTenDetails from "../components/TopTenDetails.vue"
 import TopTenAccordion from "../components/TopTenAccordion.vue"
 import SystemScoreSection from "../components/SystemScoreSection.vue"
 import downloadjs from "downloadjs";
+import { type Technique } from "../data/DataTypes"
 
 export default defineComponent({
     components: { TopTenSidebar, TopTenDetails, TopTenAccordion, SystemScoreSection },
     data() {
         return {
             calculatorStore: useCalculatorStore(),
             activeItemId: 0,
+            rankedList: Array<Technique>
         };
     },
     computed: {
-        rankedList() {
-            return this.calculatorStore.techniques
-        }
+        filters() {
+            return this.calculatorStore.activeFilters;
+        },
+        scores() {
+            return this.calculatorStore.systemScoreObj;
+        },
     },
     methods: {
-        setActiveIndex(index) {
+        setActiveIndex(index: number) {
             this.activeItemId = index
         },
-        deleteTechnique(index) {
-            this.calculatorStore.removeTechnique(index)
+        deleteTechnique(index: number) {
+            this.rankedList.splice(index, 1)
             if (index < this.activeItemId) {
                 this.setActiveIndex(this.activeItemId - 1)
             }
         },
         download() {
             downloadjs(JSON.stringify(this.rankedList.slice(0, 10)), "TopTenTechniques.json", JSON)
+        },
+        setRankedList() {
+            let filteredList = JSON.parse(JSON.stringify(this.calculatorStore.techniques));
+            filteredList = this.applyScores(filteredList)
+            filteredList = this.applyFilters(filteredList)
+            filteredList.sort(
+                (a, b) => b.adjusted_score - a.adjusted_score
+            );
+            this.rankedList = filteredList
+        },
+        applyFilters(filteredList: Array<Technique>) {
+            const newFilterList = []
+            let filterByCIS = true;
+            let filterByNIST = true;
+            if ((this.filters.nist.length === 0 || this.filters.nist.length === this.calculatorStore.allNISTOptions.length)) {
```

Comment:
Once again, following the heuristic outlined above, you can simply check if the `Set` is empty.